src/aliases/store/models/alias.go

```diff
@@ -15,29 +22,64 @@ type Alias struct {
 	Value AliasValue
 }
 
-func AliasFromEntity(ent aliasent.Alias) Alias {
+func checkAliasValue(value aliasent.AliasValue) error {
+	for _, v := range value {
+		// we just want the decoding to happen without error
+		_, err := base64.StdEncoding.DecodeString(v)
+		if err != nil {
+			return errors.InvalidFormatError("alias value is incorrect: %v: %v", err, v)
+		}
+	}
+	return nil
+}
+
+func AliasFromEntity(ent aliasent.Alias) (alias Alias, err error) {
+	// check that the ent.Value only contains base64 string
+	err = checkAliasValue(ent.Value)
+	if err != nil {
+		return alias, err
+	}
+
+	var b bytes.Buffer
+	err = json.NewEncoder(&b).Encode(ent.Value)
```

Comment:
if I understand correctly, here you are transforming `value` into its JSON representation. If that's correct then this operation here is unnecessary.

Reply:
How come?

Reply:
go-pg driver automatically marshall Go struct to its JSON representation if the destination is a TEXT/JSON in the DB.

I removed it.